parser/regexp.go

```diff
@@ -0,0 +1,100 @@
+package parser
+
+import (
+	"fmt"
+	"regexp"
+	goregexp "regexp"
+	"strings"
+
+	regexp2 "github.com/dlclark/regexp2"
+)
+
+type Regexp struct {
+	expression string
+	regexpGo *goregexp.Regexp
+	regexpLib *regexp2.Regexp
+}
+
+func NewRegexp(re string) *Regexp {
+	r := &Regexp{expression: re}
+	r.MustCompile()
+	return r
+}
+
+func (r *Regexp) compileGoRegex() error {
+		// Make some adjustments for a different regex engine than upstream matomo
+		rg := r.expression
+		rg = strings.Replace(rg, `/`, `\/`, -1)
+		rg = strings.Replace(rg, `++`, `+`, -1)
+		rg = strings.Replace(rg, `\_`, `_`, -1)
+		// if we find `\_` again, the original was `\\_`,
+		// so restore that so the regex engine does not attempt to escape `_`
+		rg = strings.Replace(rg, `\_`, `\\_`, -1)
+
+		str := `(?i)(?:^|[^A-Z0-9-_]|[^A-Z0-9-]_|sprd-)(?:` + rg + ")"
+
+		var err error
+		r.regexpGo, err = regexp.Compile(str)
+		return err
+}
+
+func (r *Regexp) compileLibRegex() error {
+		// Make some adjustments for a different regex engine than upstream matomo
+		rg := r.expression
+		rg = strings.Replace(rg, `/`, `\/`, -1)
+		rg = strings.Replace(rg, `++`, `+`, -1)
+		rg = strings.Replace(rg, `\_`, `_`, -1)
+		// if we find `\_` again, the original was `\\_`,
+		// so restore that so the regex engine does not attempt to escape `_`
+		rg = strings.Replace(rg, `\_`, `\\_`, -1)
+
+		str := `(?:^|[^A-Z0-9-_]|[^A-Z0-9-]_|sprd-)(?:` + rg + ")"
+
+		var err error
+		r.regexpLib, err = regexp2.Compile(str, regexp2.IgnoreCase)
+		return err
+}
+
+func (r *Regexp) Compile() error {
+	if r.regexpGo == nil && r.regexpLib == nil {
+		if err := r.compileGoRegex(); err != nil {
+			return r.compileLibRegex()
+		}
+	}
+	return nil
+}
+
+func (r *Regexp) MustCompile() {
+	if err := r.Compile(); err != nil {
+		panic(fmt.Sprintf("Could not compile regex '%s': %v", r.expression, err))
```

Comment:
You can simplify this with `panicf`, though panicking is a bit cringe.
```suggestion
		panicf("Could not compile regex '%s': %v", r.expression, err)
```

Reply:
Weird, `panicf` is giving me compiler errors. Is it built-in?

Yeah panicking seems less than ideal, but it matches previous behavior. And it's a bit of work to refactor everything w/ proper error handling. But I could do that in another pr.

Reply:
I'm seeing` log.Panicf` other places in the codebase https://github.com/muxinc/mux/blob/576ffa03453e40cee82b4e0ee17908b2b3635ffe/core/golang/log/log.go#L106